droid-api/src/main/java/uk/gov/nationalarchives/droid/internal/api/DroidAPI.java

```diff
@@ -192,7 +223,8 @@ public DroidAPI build() throws SignatureParseException {
             String containerVersion = StringUtils.substringAfterLast(containerSignature.getFileName().toString(), "-").split("\\.")[0];
             String droidVersion = ResourceBundle.getBundle("options").getString("version_no");
             ContainerApi containerApi = new ContainerApi(droidCore, containerSignature);
-            return new DroidAPI(droidCore, containerApi.zipIdentifier(), containerApi.ole2Identifier(), containerApi.gzIdentifier(), containerVersion, droidCore.getSigFile().getVersion(), droidVersion, this.s3Client, this.httpClient, this.s3Region);
+            List<HashAlgorithm> hashAlgorithmsOrEmptyList = this.hashAlgorithms == null ? Collections.emptyList() : this.hashAlgorithms;
```

Comment:
This might be my ignorance of Java, but on line 182, couldn't you just do

`private List<HashAlgorithm> hashAlgorithms = Collections.emptyList();` or 
`private List<HashAlgorithm> hashAlgorithmsOrEmptyList = Collections.emptyList();`?

Reply:
I think you're right, I've changed the builder so it calls `private List<HashAlgorithm> hashAlgorithms = Collections.emptyList();`

Reply:
You would still need the null protection in case someone directly uses `Builder` and passes in a null. Might be easier to null check in `public DroidAPIBuilder hashAlgorithms(final List<HashAlgorithm> hashAlgorithms)`

Reply:
I did think that but then I thought, if you're explicitly passing null into a builder then you deserve what happens to you. That being said, I've added a null check in the builder method.